Merging Master and Develop

aanchirinah opened the pull request:
#### What does this PR do?
This PR is for a peer review

#### Description of Task to be completed?
N/A

#### How should this be manually tested?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
N/A
#### Screenshots (if appropriate)

#### Questions:

aanchirinah committed:
Merge pull request #7 from andela-aanchirinah/feature/147157375/API-User-should-be-able-to-add-more-roles

#147157375 API User should be able to add more roles

aanchirinah committed:
chore(#146922329): Setup Webpack Task Runner
- setup development and production configurations for webpack
[delivers #146922329]

aanchirinah committed:
chore(#146922329): Setup Webpack Task Runner
- fixed test script
[delivers #146922329]

aanchirinah committed:
chore(#146922329): fixed test coverage script
[delivers #146922329]

aanchirinah committed:
Merge pull request #8 from andela-aanchirinah/chore/146922329/Setup-Webpack-Task-Runner

#146922329 Setup Webpack Task Runner

aanchirinah committed:
feat(SignIn): Implement user sign in into app
- Created a landing page to provide form for sign in
- Added a reducer to handle authentications
- Added methods to restrict routes
[delivers #147423645]

aanchirinah committed:
feat(SignUp): Implement user sign up
- Created a sign up form modal
- added materialize ui sass files
[delivers #147423645]

aanchirinah committed:
Merge pull request #9 from andela-aanchirinah/feature/147423645/User-should-be-able-signin-or-signup

#147423645 User should be able to sign in or sign up

aanchirinah committed:
feat(#147424013): Signed In user should see all public documents
- Created action for fetching documents
- Created reducers for handling documents state
- Created a document card component to render each document instance
[delivers #147424013]

aanchirinah committed:
feat(#147807999): Implement view all users
- created a table view to list all users
- created actions and reducer for handling user information fetching
[delivers #147807999]

aanchirinah committed:
Merge pull request #10 from andela-aanchirinah/feature/147424013/User-should-see-public-documents-once-signed-in

#147424013 User should be able to view all public documents once signed in

aanchirinah committed:
feat(#147424063): Create Document
- include quill library to create document editor
- create action for saving document
- create reducer to handle create document action
- mapped dispatch to props
- validate before saving document
[delivers #147424063]

aanchirinah committed:
Merge pull request #11 from andela-aanchirinah/feature/147424063/Signed-in-User-should-be-able-to-create-a-document

#147424063): Create Document

aanchirinah committed:
feat(#147424115): Users should be able to edit document
- created modals for viewing and editting document
- created modals to confirm delete
[delivers #147424115]

aanchirinah committed:
feat(#147424115): User can edit saved documents
- created a component modal to popup for validation
[delivers #147424115]

aanchirinah committed:
feat(#147424081): User can delete document
- create confirmation modal for delete
- create action and reducer for delete
- create generic snackbar for executed actions
[delivers #147424081]

aanchirinah committed:
Merge pull request #12 from andela-aanchirinah/feature/147424115/Signed-in-user-should-be-able-to-update-document

#147424115 Signed in user should be able to update document

aanchirinah committed:
fix(react-router-dom): Blank pages after refresh now fixed
- Had to use HashRouter to resolve the history conflicts
[fixes #147873207]

aanchirinah committed:
Merge pull request #13 from andela-aanchirinah/fix/147873207/Refreshing-pages-after-signup-makes-page-blank

#147873207 Refreshing pages after signin or signup makes them blank

aanchirinah committed:
feat(#148046303): Implement SignOut
- added a signout confirmation modal
[delivers #148046303]

aanchirinah committed:
feat(#148046303): User can signout
- created a modal for user to confirm signout
- clear user persistance data in localstorage after signout
[delivers #148046303]

aanchirinah committed:
Merge pull request #14 from andela-aanchirinah/feature/148046303/User-should-be-able-to-signout-successfully

#148046303 User should be able to signout successfully

aanchirinah committed:
chore(#148091801): Deploy to Heroku

aanchirinah committed:
chore(#148091801): Deploy to Heroku
- configured production env values
[delivers #148091801]

aanchirinah committed:
Merge pull request #16 from andela-aanchirinah/chore/148091801/Deploy-to-Heroku

#148091801 Deploy to heroku

aanchirinah committed:
feat(#148091781): Add pagination
- included pagination with interfaces for browsing though pages
[delivers #148091781]

aanchirinah committed:
Merge pull request #17 from andela-aanchirinah/feature/148091781/Documents-list-should-be-paginated

#148091781 Documents list should be paginated

aanchirinah committed:
feat(#148188963): Paginate User lists
- User lists paginated
[delivers #148188963]

aanchirinah committed:
feat(#148188963): Paginate User lists
- Added JSDOC to all functions
[delivers #148188963]

aanchirinah committed:
Merge pull request #18 from andela-aanchirinah/feature/148188963/Paginate-user-lists

#148188963 Paginate user lists

aanchirinah committed:
chore(#148223367): Setup unit testing for frontend
- install jest, moxioxs and redux-mock-store packages
- configured jest.config.js
[finished #148223367]

aanchirinah committed:
Merge pull request #20 from andela-aanchirinah/chore/148223367/Setup-unit-testing-for-frontend

#148223367 Setup unit testing for frontend

aanchirinah committed:
chore(14809180): Deploy to heroku:
- Fixed Heroku issue in package.json
[delivers #14809180]

aanchirinah committed:
chore(14809180): Deploy to heroku:
- Fixed Heroku issue in package.json
[delivers #14809180]

aanchirinah committed:
chore(14809180): Deploy to heroku:
- Fixed Heroku issue in package.json
[delivers #14809180]

aanchirinah committed:
chore(14809180): Deploy to heroku:
- Fixed Heroku issue in package.json
[delivers #14809180]

aanchirinah committed:
chore(14809180): Deploy to heroku:
- Fixed Heroku issue in package.json
[delivers #14809180]

aanchirinah committed:
chore(14809180): Deploy to heroku:
- Fixed Heroku issue in package.json
[delivers #14809180]

aanchirinah committed:
add dependencies to package.json

aanchirinah committed:
fix heroku build

aanchirinah committed:
feat(#147424097): User should be able to edit their profile
- created modal for editing user profile
- added a confirm password field
[delivers #147424097]

aanchirinah committed:
feat(#147424097): User should be able to edit their profile
- created modal for editing user profile
- added a confirm password field
[delivers #147424097]

aanchirinah committed:
Merge pull request #22 from andela-aanchirinah/feature/147424097/Signed-in-user-should-be-able-to-edit-their-profile

#147424097 Signed in user should be able to edit their profile

aanchirinah committed:
feat(#147424173): Admin can assign roles to users
- added an edit control on the user list to edit user role
[delivers #147424173]

aanchirinah committed:
Merge pull request #23 from andela-aanchirinah/feature/147424173/Admin-can-assign-users-roles

#147424173 Admin can assign users roles

desaawa commented on client/constants/Constants.jsx:
```diff
@@ -0,0 +1,38 @@
+export const AUTH = {
```
I believe this file doesn't have anything to do with JSX, so I should be a `.js` file

desaawa commented on client/actions/AuthActions.jsx:
```diff
@@ -0,0 +1,83 @@
+        });
+      });
+  };
+}
```
This file should not end in `.jsx` since it is not a JSX file. Use `.js` for actions

desaawa commented on client/store/configureStore.jsx:
```diff
@@ -0,0 +1,32 @@
+  return store;
+};
+
+export default configureStore;
```
This is not a JSX file, and so should not end in `.jsx` but `.js`

desaawa reviewed (changes requested):
On the frontend, Actions, Reduce, Constants, Utilities should not end in `.jsx` but in `.js`
On the backend, you named your `documents` controllers and routes in plural form but didn't do same for role and user. They should all be pluralized.

Nice work in general man!

aanchirinah committed:
chore(#148472107): Setup E2E testing
- written tests for components
[delivers #148472107]

aanchirinah committed:
chore(#148472107): Setup E2E testing
- install selenium and nightwatch
- configure nightwatch.json
- wriiten some e2e test for landing page
[delivers #148472107]

aanchirinah committed:
chore(#148472107): Setup E2E testing
- install selenium and nightwatch
- configure nightwatch.json
- wriiten some e2e test for landing page
[delivers #148472107]

aanchirinah committed:
Merge pull request #24 from andela-aanchirinah/chore/148472107/Setup-E2E-tesing-for-frontend

#148472107 Setup e2e tesing for frontend

aanchirinah committed:
fix(#148548851): Fixed password for sign in and sign up
[fixes #148548851]

aanchirinah committed:
Merge pull request #25 from andela-aanchirinah/fix/148548851/Fix-password-length

#148548851 Fix password length

aanchirinah committed:
feat(feedback): Peer review feedback implementation
- renamed some .jsx files to .js
- included indicating role access to creating new document
[delivered]

aanchirinah committed:
Merge pull request #26 from andela-aanchirinah/peer-feedback

Peer review feedback implementation

aanchirinah committed:
Cleared git cache

aanchirinah committed:
feat(Feeback): Peer review feedback
- implement peer feedback
[delivers]

aanchirinah committed:
Merge pull request #27 from andela-aanchirinah/peer/review-2

Peer/review 2

aanchirinah committed:
bug(#148656875): Fixed checkpoint defence bugs
- Fixed all the bugs and feedback from checkpoint defence
- More details in pull request
[fixes #148656875]

aanchirinah committed:
Merge pull request #28 from andela-aanchirinah/bug/148656875/Fix-checkpoint-defence-bugs

#148656875  Fixed checkpoint defence bugs

aanchirinah committed:
chore(148894253): Write more component tests
- Added more component test for client
- added more redux tests
[delivers #148894253]

aanchirinah committed:
Merge pull request #29 from andela-aanchirinah/chore/148894253/Write-more-component-tests

#148894253 Write more component tests

nelsonabienorin commented on client/actions/AuthActions.js:
```diff
@@ -0,0 +1,83 @@
+ * @param  {string} type  [action type description]
+ * @return {object}       [action object with type and payload]
+ */
+export function login(token, type) {
```
You may want  to go for the ES6 way of defining a function

nelsonabienorin commented on client/components/Main.jsx:
```diff
@@ -0,0 +1,29 @@
+
+injectTapEventPlugin();
+
+
```
A single line space can be applied

nelsonabienorin commented on client/components/Main.jsx:
```diff
@@ -0,0 +1,29 @@
+import injectTapEventPlugin from 'react-tap-event-plugin';
+
+
+injectTapEventPlugin();
```
What does this method `injectTapEventPlugin` do?

nelsonabienorin commented on client/components/dashboard/AdminDocumentList.jsx:
```diff
@@ -0,0 +1,156 @@
+ * @return {null}       returns nothing
+ */
+  componentDidMount() {
+    // disptching action to fetch documents here
```
Reword your typo to dispatch

nelsonabienorin commented on client/components/dashboard/DashboardDrawer.jsx:
```diff
@@ -0,0 +1,86 @@
+  );
+};
+
+
```
Single line spacing could be better

nelsonabienorin commented on client/components/landing/SignIn.jsx:
```diff
@@ -0,0 +1,291 @@
+  /**
+   * This function pops up a snackbar to alert user of wrong password input
+   * during signup
+   * @return {null}       retruns nothing
```
retruns or returns?

nelsonabienorin commented on client/components/modals/EditDocumentModal.jsx:
```diff
@@ -0,0 +1,283 @@
+  /**
+   * This function changes intial states based on onChange events
+   * @param  {object} event [the events object parameter]
+   *@return {null}       retruns nothing
```
Typo and indentation issues here

nelsonabienorin commented on client/components/modals/EditDocumentModal.jsx:
```diff
@@ -0,0 +1,283 @@
+
+  /**
+   * Checks props.access value to set initial checkbox state
+   * @return {null}       retruns nothing
```
kindly check all instances of this typo in your comment

nelsonabienorin commented on client/components/modals/EditDocumentModal.jsx:
```diff
@@ -0,0 +1,283 @@
+    };
+    return (
+      <div>
+        <Dialog
```
This block can be indented in a better way

nelsonabienorin commented on client/reducers/DocumentReducer.js:
```diff
@@ -0,0 +1,68 @@
+    newState = [Object.assign({}, newDoc), ...state];
+    if (newState.length > 9) newState.pop();
+    return newState;
+
```
Can the line spacing be consistent?

nelsonabienorin commented on client/reducers/DocumentReducer.js:
```diff
@@ -0,0 +1,68 @@
+  switch (action.type) {
+  case DOCUMENT.GET_SUCCESS:
+    return action.document;
+
```
line spacing?

nelsonabienorin commented on client/styles/app.scss:
```diff
@@ -0,0 +1,56 @@
+// @charset "UTF-8";
```
These commented block can be gotten rid of if is not needed

nelsonabienorin commented on client/styles/components/_CardHeader.scss:
```diff
@@ -0,0 +1,24 @@
+.CustomCard {
+  // span{
```
Unnecessary comments can be gotten rid of

nelsonabienorin commented on client/styles/components/_sideNav.scss:
```diff
@@ -0,0 +1,214 @@
+  .collapsible {
+    margin: 0;
+  }
+
```
Can the line spacing be more consistent?

nelsonabienorin reviewed (commented):
Good job done so far, kindly effect the changes requested and respond where applicable.

aanchirinah committed:
chore(#148958247): Write more E2E tests
- Added more end to end tests for client
[delivers #148958247]

aanchirinah committed:
Merge pull request #30 from andela-aanchirinah/chore/148958247/Write-more-e2e-tests

#148958247 Write more E2E tests

aanchirinah committed:
feat(#148958251): Setup API Documentation with Swagger API
- Complete API Documentation with swagger
[delivers #148958251]

aanchirinah committed:
Merge pull request #31 from andela-aanchirinah/feat/148958251/Develop-documentation-with-Swagger-API

#148958251 Setup API Documentation with Swagger API

aanchirinah committed:
feat(Responsiveness): Implement peer review
[delivers]

aanchirinah committed:
feat(Peer-review): Implement peer review feedback
- Fixed all reviews
[delivers]

aanchirinah committed:
Merge pull request #32 from andela-aanchirinah/feat/implement/peer-review-4

Peer Review 4

aanchirinah committed:
feat(feedback): Implement feedback
- Implement all feedback form last checkpoint defence
[delivers]

aanchirinah committed:
Merge pull request #34 from andela-aanchirinah/implement-feedback

Implement Feedback Part 2

aanchirinah committed:
feat(scorecard-feedback): Implementing ScoreCard Feedback
- protect password display in seeded files
- Fix footer
- Change 'Yaayyy' to 'Success'
- DocumentCard should follow DRY principle
[delivered]

aanchirinah committed:
feat(feedback): Implement feedback from scorecard
- remove unused parameters and functions
[delivers]

aanchirinah committed:
Merge pull request #35 from andela-aanchirinah/feat/implement-feedback-4

Implementing ScoreCard Feedback

aanchirinah committed:
fix(tests): Fix component test typographical errors
[fixes]

aanchirinah committed:
feat(component-tests): Implement feedback on scorecard
- Refactor test descriptions to make it more reading friendly
[delivers]

aanchirinah committed:
feat(tests): Implement Scorecard Feedback
- Fixed all exceed 80 characters limit eslint flags
[fixed]

aanchirinah committed:
Merge pull request #36 from andela-aanchirinah/implement-scorecard-feedback-part-2

fix(tests): Fix component test typographical errors

aanchirinah committed:
Fixed jquery vulnerability issue

aanchirinah committed:
Merge pull request #37 from aanchirinah/Fix-dependency-issue

Fixed jquery vulnerability issue
